strings/base_windows.h

```diff
@@ -128,7 +128,11 @@ namespace winrt::impl
         }
 
         void* result{};
-        ptr->QueryInterface(guid_of<To>(), &result);
+        hresult code = ptr->QueryInterface(guid_of<To>(), &result);
+        if (code < 0)
+        {
+            WINRT_IMPL_RoCaptureErrorContext(code);
```

Comment:
Is this still needed now that `WINRT_IMPL_SHIM` isn't being used above?

Reply:
Yes it is needed.  There is no HRESULT return from this method so the COM error context is used to propagate the failure code upwards to `check_cast_result`.